packages/features/bookings/lib/handleNewBooking.ts

```diff
@@ -2515,17 +2516,29 @@ async function handler(
             calEvent: getPiiFreeCalendarEvent(evt),
           })
         );
-        await sendScheduledEmails(
-          {
-            ...evt,
-            additionalInformation: metadata,
-            additionalNotes,
-            customInputs,
-          },
-          eventNameObject,
-          isHostConfirmationEmailsDisabled,
-          isAttendeeConfirmationEmailDisabled
-        );
+        const client = await getClient();
+        const handle = await client.workflow.start(sendScheduledEmails, {
+          taskQueue: "calcom", //TODO: this needs to come from a shared config.
+          // type inference works!
+          args: [
+            {
+              data: {
+                calEvent: {
+                  ...evt,
+                  additionalInformation: metadata,
+                  additionalNotes,
+                  customInputs,
+                },
+                eventNameObject,
+                hostEmailDisabled: isHostConfirmationEmailsDisabled,
+                attendeeEmailDisabled: isAttendeeConfirmationEmailDisabled,
+              },
+            },
+          ],
+          // TODO: can we use a meaningful business ID, like eventId instead of uuidv4
+          workflowId: `scheduled-email-${crypto.randomUUID()}`,
+        });
```

Comment:
This is where the Workflow Execution is triggered using the newly implemented system. The only change needed to use the system is to swap the direct calls to the email-related functions to a similar workflow call.